src/test/java/com/savvato/tribeapp/services/ReviewDecisionServiceImplTest.java

```diff
@@ -0,0 +1,79 @@
+package com.savvato.tribeapp.services;
+
+import com.savvato.tribeapp.controllers.dto.ReviewDecisionRequest;
+import com.savvato.tribeapp.entities.ReviewDecision;
+import com.savvato.tribeapp.entities.ReviewDecisionReason;
+import com.savvato.tribeapp.repositories.ReviewDecisionReasonRepository;
+import com.savvato.tribeapp.repositories.ReviewDecisionRepository;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.ArgumentCaptor;
+import org.mockito.Mockito;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.context.TestConfiguration;
+import org.springframework.boot.test.mock.mockito.MockBean;
+import org.springframework.context.annotation.Bean;
+import org.springframework.dao.DataAccessException;
+import org.springframework.test.context.junit.jupiter.SpringExtension;
+
+import java.util.Optional;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.mockito.Mockito.times;
+import static org.mockito.Mockito.verify;
+
+@ExtendWith(SpringExtension.class)
+
+public class ReviewDecisionServiceImplTest {
+    @TestConfiguration
+    static class ReviewDecisionServiceTestContextConfiguration {
+        @Bean
+        public ReviewDecisionService reviewDecisionService() {
+            return new ReviewDecisionServiceImpl();
+        }
+    }
+
+    @Autowired
+    ReviewDecisionService reviewDecisionService;
+
+    @MockBean
+    ReviewDecisionRepository reviewDecisionRepository;
+    @MockBean
+    ReviewDecisionReasonRepository reviewDecisionReasonRepository;
+
+    // TODO: Confirm no need to test for reviewerId, reviewId being invalid since those should be automatically generated
+    // TODO: Confirm length of test names
```

Comment:
Re: length of test names: Err on the side of being too long.. be verbose.

Reply:
Thank you! I have removed this comment.